### PR TITLE
Translations entity -- internal representation

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR relates to processing of entities from translations.txt. The goal is to use the required use cases implemented via PR #44 that allow to go from a row of translations.txt to a specific concrete internal representation.

**Expected behavior:** 

Explain and/or show screenshots for how you expect the pull request to work in your testing (in case other devices exhibit different behavior).

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] ~Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")~
- [x] Linked all relevant issues
- [ ] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedTranslationTest.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        inOrder.verify(mockBuilder, times(1)).build();
+
+        inOrder.verify(mockEntityBuildResult, times(1)).isSuccess();
+
+        // suppressed warning regarding unused result of method, since this behavior is wanted
+        //noinspection ResultOfMethodCallIgnored
+        inOrder.verify(mockEntityBuildResult, times(1)).getData();
+
+        inOrder.verify(mockGtfsDataRepo, times(1)).addTranslation(mockTranslation);
+
```
Comment (maximearmstrong) [inline usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedTranslationTest.java:92]: Maybe we could add a `verifyNoInteraction` with `mockResultRepo` here

File: domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/translations/TranslationTest.java [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    private static final String VALID_TABLE_NAME = "agency";
+    private static final String FIELD_NAME = "field name";
+    private static final String LANGUAGE = "language";
+    private static final String TRANSLATION = "translation";
+    private static final String DEFINED_RECORD_ID = "record id";
+    private static final String DEFINED_RECORD_SUB_ID = "record sub id";
+    private static final String DEFINED_FIELD_VALUE = "field value";
+    static final String UNDEFINED_FIELD_VALUE = null;
+    static final String UNDEFINED_RECORD_SUB_ID = null;
+    static final String UNDEFINED_RECORD_ID = null;
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/translations/TranslationTest.java:41]: They should be, modifying this.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/translations/Translation.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                if (tableName != TableName.FEED_INFO && ((recordId != null && fieldValue != null) ||
+                        (recordId == null && fieldValue == null))){
+                    noticeCollection.add(new IllegalFieldValueCombination("translations.txt",
+                            "record_id", "field_value", null));
+                }
+                if (recordSubId != null && fieldValue != null) {
+                    noticeCollection.add(new IllegalFieldValueCombination("translations.txt",
+                            "record_sub_id", "field_value", null));
+                }
+                if (recordSubId == null && tableName!= null && tableName==TableName.STOP_TIMES && recordId != null) {
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/translations/Translation.java:260]: Good catch!

File: adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java [hunk 4]
```diff
@@ -557,4 +559,66 @@ void addAttributionAndGetAttributionShouldReturnSameEntity() {
...
+
+        underTest.addTranslation(mockTranslation);
+
+        assertNull(underTest.addTranslation(mockTranslation));
+    }
+
+    @Test
+    void addTranslationWithSameDataShouldReturnNul() {
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        final Translation mockTranslation00 = mock(Translation.class);
```
Comment (fabrice-v) [inline adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java:587]: 00 seems superfluous

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/DuplicatedEntityNotice.java [hunk 5]
```diff
@@ -31,6 +31,31 @@ public DuplicatedEntityNotice(final String filename, final String fieldName, fin
         putNoticeSpecific(KEY_FIELD_NAME, fieldName);
     }
 
+    public DuplicatedEntityNotice(final String filename, final String fieldName, final String entityId,
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/DuplicatedEntityNotice.java:34]: Needs javadoc to clarify parameters

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java [hunk 6]
```diff
@@ -33,6 +33,33 @@ public IllegalFieldValueCombination(final String filename, final String fieldNam
...
+    public IllegalFieldValueCombination(final String filename, final String fieldName,
+                                        final String conflictingFieldName, final String entityId,
+                                        final String compositeKeyFirstPart, final String compositeKeySecondPart,
+                                        final String compositeKeyThirdPart, final String compositeKeyFourthPart,
+                                        final Object compositeKeyFirstValue, final Object compositeKeySecondValue,
+                                        final Object compositeKeyThirdValue, final Object compositeKeyFourthValue) {
+        super(filename, E_019,
+                "Conflicting field values",
+                "Conflicting field values for fields:`" + fieldName + "` and field:`" + conflictingFieldName
+                        + "` for entity with composite id:" + System.lineSeparator() +
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java:45]: Not sure about the System.lineSeparator. Our JSON output should not contain formatting characters, so they should not be used here, or stripped before the  output. Though I'd be inclined to simply never put them in in the first place.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java [hunk 7]
```diff
@@ -33,6 +33,33 @@ public IllegalFieldValueCombination(final String filename, final String fieldNam
         putNoticeSpecific(KEY_CONFLICTING_FIELD_NAME, conflictingFieldName);
     }
 
+    public IllegalFieldValueCombination(final String filename, final String fieldName,
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java:36]: Should be postfixed with `Notice`

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingRequiredValueNotice.java [hunk 8]
```diff
@@ -32,6 +32,32 @@ public MissingRequiredValueNotice(String filename, String fieldName, String enti
...
+    public MissingRequiredValueNotice(final String filename, final String fieldName, final String entityId,
+                                      final String compositeKeyFirstPart, final String compositeKeySecondPart,
+                                      final String compositeKeyThirdPart, final String compositeKeyFourthPart,
+                                      final String compositeKeyFirstValue, final String compositeKeySecondValue,
+                                      final String compositeKeyThirdValue, final String compositeKeyFourthValue){
+        super(filename, E_015,
+                "Missing required value",
+                "Missing value for field:`" + fieldName
+                        + "` marked as required in entity with composite id:`" + System.lineSeparator() +
+                        "`" + compositeKeyFirstPart + "`: " + compositeKeyFirstValue + "`" + System.lineSeparator() +
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingRequiredValueNotice.java:44]: Remove `System.lineSeparator`

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/UnexpectedEnumValueNotice.java [hunk 9]
```diff
@@ -33,6 +33,35 @@ public UnexpectedEnumValueNotice(final String filename, final String fieldName,
...
+    public UnexpectedEnumValueNotice(final String filename, final String fieldName, final String entityId,
+                                     final Object enumValue,
+                                     final String compositeKeyFirstPart, final String compositeKeySecondPart,
+                                     final String compositeKeyThirdPart, final String compositeKeyFourthPart,
+                                     final Object compositeKeyFirstValue, final Object compositeKeySecondValue,
+                                     final Object compositeKeyThirdValue, final Object compositeKeyFourthValue) {
+        super(filename, E_021,
+                "Unexpected enum value",
+                "Invalid value :`" + enumValue + "` - for field:`" + fieldName + "` in file:`" + filename +
+                        "` for entity with composite id:`" + System.lineSeparator() +
```
Comment (fabrice-v) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/UnexpectedEnumValueNotice.java:45]: Remove `System.lineSeparator`

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java (hunk 6 above)
Comment (lionel-nj) [reply domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/IllegalFieldValueCombination.java:45]: As discussed, this will be removed.

File: usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java [hunk 10]
```diff
@@ -86,7 +86,7 @@ public void execute(final ParsedEntity validatedParsedStopTime) {
         if (stopTime.isSuccess()) {
             if (gtfsDataRepository.addStopTime((StopTime) stopTime.getData()) == null) {
                 resultRepository.addNotice(new DuplicatedEntityNotice("stop_times.txt", null,
-                        validatedParsedStopTime.getEntityId(), "trip_id",
+                        "trip_id",
```
Comment (fabrice-v) [inline usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java:89]: Seems unrelated. Also, ecah parameter should be on their own line

File: usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTimeTest.java [hunk 11]
```diff
@@ -262,10 +262,6 @@ void duplicateParsedStopTimeShouldGenerateNoticeAndNotBeAddedToGtfsDataRepo() {
 
         verify(mockDataRepo, times(1)).addStopTime(mockStopTime);
 
-        //result of method .getEntityId() is not used here, therefore removing this warning to avoid lint
-        //noinspection ResultOfMethodCallIgnored
-        verify(mockParsedEntity, times(1)).getEntityId();
```
Comment (fabrice-v) [inline usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTimeTest.java:267]: Seems unrelated

File: usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java (hunk 10 above)
Comment (lionel-nj) [reply usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java:89]: The constructor of `DuplicatedEntityNotice` that is used here has been modified, therefore this change might not seem related to the PR but is needed.

File: usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTimeTest.java (hunk 11 above)
Comment (lionel-nj) [reply usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTimeTest.java:267]: see https://github.com/MobilityData/gtfs-validator/pull/90/files/279e29b9681823099ca9e965d1f4eec9acdc0f83#r450924854

File: usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java (hunk 10 above)
Comment (fabrice-v) [reply usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ProcessParsedStopTime.java:89]: Oh, alright. That make sense
